### PR TITLE
Add AddLink method to DummySpan

### PR DESCRIPTION
**Pull request by vikmik**
This is only in case you'd rather not merge #191 because of the minimum go version bump.

As it stands, projects with holster as indirect dependency cannot build if they also use otel v1.25.0. See also #189 
For example, a project using gubernator and otel 1.25.0 cannot build at the moment, because of this.

OTel added a new function to the Span interface, so this PR is the minimal change that can unblock things. I would prefer #191 to be merged, but either will work.

### CODE REVIEW

File: tracing/dummy_span.go
Comment (vtopc) [inline tracing/dummy_span.go:72]: Please, run gofmt/[goimports](https://pkg.go.dev/golang.org/x/tools/cmd/goimports)

**Review by Baliedge — APPROVED** [review]
I opt to approve this PR over #191 to solve the immediate need while avoiding the OTel upgrade complexities, which is a separate issue.

**Comment by vikmik** [comment]
@vtopc @Baliedge sorry for the linting mishap - I've updated the PR, could you retrigger the workflows?

**Review by vtopc — APPROVED** [review]

**Comment by vikmik** [comment]
Thanks @vtopc - once this PR is merged it will unblock our project

**Comment by vtopc** [comment]
> Thanks @vtopc - once this PR is merged it will unblock our project

No. Thank you.

**Pull request merged**
